test/clj/puppetlabs/trapperkeeper/testutils/webserver.clj

```diff
@@ -55,3 +58,25 @@
      ~port-var
      {}
      ~@body))
+
+(defn get-jetty-mbean-object-names
+  "TODO SPLAIN"
+  []
+  (let [mbeanserver (ManagementFactory/getPlatformMBeanServer)]
+    (.queryNames mbeanserver
+                 nil
+                 (ObjectName. "org.eclipse.jetty.jmx:type=mbeancontainer,*"))))
+
+(defn assert-clean-shutdown
+  "TODO SPLAIN"
+  [f]
+  (f)
+  ;; This sucks, because if this assertion fails, it will not give any clue as to
+  ;; which test caused it to fail beyond just the test namespace.  However, I tried
+  ;; several things (such as throwing an exception here rather than having an assertion),
+  ;; and nothing gave any better debugging info because the metadata about which
+  ;; test we're running and the call stack for the test function itself are simply
+  ;; not available from inside a fixture.  So I decided that even those these aren't
+  ;; super fun to debug, it was better than any other option in terms of enforcing
+  ;; that the tests clean up after themselves properly.
```

Comment:
If we did it as `:once`, though, is it possible that one or more bad tests in the namespace don't clean themselves up but if the last one run does we might not assert at the end?  If so, I think `:each` might still be better.  Hopefully the call to get the `mbean-names` would be cheap enough that we're not adding significant performance overhead to each test by doing this?